keras/src/backend/jax/core.py

```diff
@@ -288,6 +312,21 @@ def __setattr__(self, name, value):
     NnxVariable.__setattr__ = __setattr__
 
 
+def check_memory_spec(init_layout, shape):
+    is_named_sharding = isinstance(init_layout, jax.sharding.NamedSharding)
+    # Check if PartitionSpec has any non-None values
+    spec = getattr(init_layout, "spec", None)
+    partition_spec = spec if spec is not None else ()
+    is_partitioned = any(dim is not None for dim in partition_spec)
+    is_partitioned = is_partitioned and is_named_sharding
+    # Size check: Allow sharding at init only for large arrays
+    # Sharding smaller arrays at init increases loading time
+    size_threshold = 250 * 1024 * 1024  # 250MB
+    array_size = np.prod(shape) * 4  # Assume 4 bytes per element
+    size_check = True if array_size >= size_threshold else False
+    return is_partitioned and size_check
```

Comment:
This would be a bit easier and more readable with early exits:
```python
if not isinstance(init_layout, jax.sharding.NamedSharding)
    return False

if all(dim is None for dim in init_layout.spec):
    return False

size_threshold = 250 * 1024 * 1024  # 250MB
array_size = np.prod(shape) * 4
return array_size >= size_threshold
```